src/Microsoft.AspNet.Mvc.Core/Internal/Routing/LinkGenerationMatch.cs

```diff
@@ -9,15 +11,50 @@ public struct LinkGenerationMatch
     {
         private readonly bool _isFallbackMatch;
         private readonly AttributeRouteLinkGenerationEntry _entry;
+        private readonly IDictionary<string, object> _routeValues;
 
-        public LinkGenerationMatch(AttributeRouteLinkGenerationEntry entry, bool isFallbackMatch)
+        public LinkGenerationMatch(
+            AttributeRouteLinkGenerationEntry entry,
+            bool isFallbackMatch,
+            IDictionary<string, object> routeValues)
         {
             _entry = entry;
             _isFallbackMatch = isFallbackMatch;
+            _routeValues = routeValues;
         }
 
         public AttributeRouteLinkGenerationEntry Entry { get { return _entry; } }
 
         public bool IsFallbackMatch { get { return _isFallbackMatch; } }
+
+        public IDictionary<string, object> RouteValues { get { return _routeValues; } }
+
+        public int ParameterCount { get {
+                var routeValues = _routeValues;
```

Comment:
Because "this" isn't available in a lambda which is in a struct. That one was news to me too.